File: common/src/main/java/net/impleri/playerskills/integration/kubejs/Registries.java
package net.impleri.playerskills.integration.kubejs;

import dev.latvian.mods.kubejs.RegistryObjectBuilderTypes;
import net.impleri.playerskills.SkillResourceLocation;
import net.impleri.playerskills.api.Skill;
import net.minecraft.core.Registry;
import net.minecraft.resources.ResourceKey;

public abstract class Registries {
    private static final ResourceKey<Registry<Skill<?>>> key = ResourceKey.createRegistryKey(SkillResourceLocation.of("skill_builders_registry"));

    public static final RegistryObjectBuilderTypes<Skill<?>> SKILLS = RegistryObjectBuilderTypes.add(key, Skill.class);
}

File: common/src/main/kotlin/net/impleri/playerskills/integrations/kubejs/Registries.kt
package net.impleri.playerskills.integrations.kubejs

import dev.latvian.mods.kubejs.RegistryObjectBuilderTypes
import net.impleri.playerskills.api.Skill
import net.impleri.playerskills.utils.SkillResourceLocation.of
import net.minecraft.resources.ResourceKey

object Registries {
  private val key = ResourceKey.createRegistryKey<Skill<*>>(of("skill_builders_registry"))

  val SKILLS = RegistryObjectBuilderTypes.add(key, Skill::class.java)
}
